src/chainstate/stacks/db/blocks.rs

```diff
@@ -5234,6 +5234,7 @@ impl StacksChainState {
         current_block: &BlockHeaderHash,
         tx: &StacksTransaction,
         tx_size: u64,
+        mempool_admission_check: TransactionAnchorMode,
```

Comment:
The "no such chain tip" and various mem pool rejection reasons are the errors relevant to this new parameter. 

There aren't any "new" errors that result from this parameter, since this function attempted to validate the given tx against the confirmed chain tip and the unconfirmed chain tip (in the case of a bad nonces error) before. Thus, I'm not making any "new" function calls, just switching the ordering.

Reply:
but, isn't the format for documenting errors (suggested by you :)), that there be a section called `#Errors` and then we actually list the class names of the errors.

Reply:
I renamed the section  to errors. The failure reasons include all of the possible errors in `MemPoolRejection`, which is captured in the return type of the function.